tests/test_raft.rs

```diff
@@ -2198,7 +2549,18 @@ fn test_raft_nodes() {
 
 #[test]
 fn test_campaign_while_leader() {
+    test_campaign_while_leader_with_pre_vote(false);
+}
+
+#[test]
+fn test_pre_campaign_while_leader() {
+    test_campaign_while_leader_with_pre_vote(true);
+}
+
+
+fn test_campaign_while_leader_with_pre_vote(pre_vote: bool) {
     let mut r = new_test_raft(1, vec![1], 5, 1, new_storage());
+    r.pre_vote = pre_vote;
```

Comment:
I think it's better to add a new function `new_raft`. Set it after initialization assumes that `pre_vote` doesn't need any initialization work.